go/libkb/link_cache.go

```diff
@@ -0,0 +1,105 @@
+// Copyright 2015 Keybase, Inc. All rights reserved. Use of
+// this source code is governed by the included BSD license.
+
+package libkb
+
+type linkIDFixed [LinkIDLen]byte
+
+// LinkCache is a cache of ChainLinks.
+// It is safe to use concurrently.
+type LinkCache struct {
+	cache map[linkIDFixed]ChainLink
```

Comment:
Hmm, maybe we don't want this thing growing without bound.  Can we periodically clean out links we haven't used in a while past a certain buffer size?